packages/dev/core/src/Misc/uvprojector.ts

```diff
@@ -0,0 +1,136 @@
+import { VertexBuffer } from "../Buffers";
+import { Ray } from "../Culling/ray";
+import { Vector3 } from "../Maths";
+import { AbstractMesh } from "../Meshes/abstractMesh";
+import { CreateBox } from "../Meshes/Builders/boxBuilder";
+import { Mesh } from "../Meshes/mesh";
+import { Scene } from "../scene";
+import { FloatArray, Nullable } from "../types";
+
+export class Uvprojector {
+    /**
+     * Project uv from a reference mesh
+     * @param target the mesh to project the uv to
+     * @param reference the reference mesh to project the uv from
+     * @param scalefact the bounding box scale for the reference mesh
+     * @param scene
+     */
+     public static ProjectUV(target: Mesh, reference: Nullable<Mesh>, scalefact: number, scene: Scene): void {
+        target.bakeCurrentTransformIntoVertices();

```

Comment:
This is something that the user may not want. We should get the data as is and then transform the vector3 using the mesh data so we do not change the target mesh